lib/jobs/sftp-job.js

```diff
@@ -67,19 +73,36 @@ function sftpJobFactory(
 
             var ssh = sshClient;
             ssh.on('ready', function() {
+                console.log('====== SftpJob.ssh: ready');
                 ssh.sftp(function(err, sftp) {
                     if (err) { reject(err); }
-                    sftp.fastPut(
+                    if (self.isPDU === "true") {
+                        console.log("==== this is an iPDU, isPDU:", self.isPDU);
+                        var readStream = fs.createReadStream(self.fileSource);
+                        var writeStream = sftp.createWriteStream(self.fileDestination);
+                        writeStream.on('close',function () {
+                            console.log( "===== file transferred succesfully" );
+                        });
+                        writeStream.on('end', function () {
+                            console.log( "===== sftp connection closed" );
```

Comment:
You should use logger.debug if you want to record something. And also, you should use a callback to handle error case. And in the callback function, reject and resolve should be called.